vendor/src/github.com/docker/libnetwork/drivers/bridge/setup_device.go

```diff
@@ -27,10 +29,15 @@ func setupDevice(config *NetworkConfiguration, i *bridgeInterface) error {
 	// was not supported before that.
 	kv, err := kernel.GetKernelVersion()
 	if err == nil && (kv.Kernel >= 3 && kv.Major >= 3) {
+		i.Link.Attrs().HardwareAddr = netutils.GenerateRandomMAC()
+		log.Debugf("Setting bridge mac address to %s", i.Link.Attrs().HardwareAddr)
```

Comment:
I'm confused, here you set the MAC, _and_ assign true to setMac.

Below setMac will still be passed to CreateBridge() as true, even though you've just set the MAC here.  It looks like some lines of separation are being violated, can this be done more cleanly?